runtime/src/main/java/com/kotcrab/vis/runtime/scene/SceneFactory.java

```diff
@@ -0,0 +1,16 @@
+package com.kotcrab.vis.runtime.scene;
+
+import com.kotcrab.vis.runtime.data.SceneData;
+import com.kotcrab.vis.runtime.util.EntityEngineConfiguration;
+
+/**
+ * You can set your descendant of this class in {@link SceneLoader} for providing custom way of Scene creation.
+ *
+ * @author dpotenko
+ */
+public class SceneFactory {
```

Comment:
This class gives to users huge opportunities in decorating WorldConfiguration and Scene. It is the best way to solve my issue. Because user can only decorate existent classes by this way and can't break your internal class logic.